stats.js

```diff
@@ -269,4 +342,84 @@ function init() {
     }
 }
 
+function showMinPayouts(minPayoutsValue) {
+  STAT_MIN_PAYOUTS_VALUE.textContent = minPayoutsValue;
+  INPUT_MIN_PAYOUTS.value = minPayoutsValue;
+}
+
+OPEN_MODAL_BTNS.forEach((btn) => {
+  btn.addEventListener('click', () => {
+    MODAL.showModal();
+  });
+});
+
+CLOSE_MODAL_BTN.addEventListener('click', () => {
+  MODAL.close();
+});
+
+MODAL.addEventListener('click', (e) => {
+  const dialogDimensions = MODAL.getBoundingClientRect();
+  if (
+    e.clientX < dialogDimensions.left ||
+    e.clientX > dialogDimensions.right ||
+    e.clientY < dialogDimensions.top ||
+    e.clientY > dialogDimensions.bottom
+  ) {
+    MODAL.close();
+  }
+});
+
+function assignFormListenerMinPayoutsForm(wallet) {
+  FORM_MIN_PAYOUTS.addEventListener('submit', handleSubmit.bind(null, wallet));
+}
+
+function handleSubmit(wallet, e) {
+  e.preventDefault();
+  const newValue = INPUT_MIN_PAYOUTS.value;
+
+  disableSubmitButton();
+
+  createUserValue(wallet, 'min_payout', newValue)
+    .then(handleSuccess)
+    .catch(handleError)
+    .finally(resetSubmitButton);
+}
+
+function disableSubmitButton() {
+  FORM_SUBMIT_BTN.disabled = true;
+  FORM_SUBMIT_BTN.classList.add('loading');
+  FORM_SUBMIT_BTN.textContent = 'Saving...';
+}
+
+function resetSubmitButton() {
+  FORM_SUBMIT_BTN.disabled = false;
+  FORM_SUBMIT_BTN.textContent = 'Save Changes';
+  FORM_SUBMIT_BTN.classList.remove('loading');
+}
+
+function handleSuccess(res) {
+  showMinPayouts(res.value);
+  ERROR_MESSAGE_ELEMENT.textContent =
+    'The minimum payout was successfully updated.';
+  INPUT_MIN_PAYOUTS.classList.remove('invalid');
+  INPUT_MIN_PAYOUTS.classList.add('success');
+  setTimeout(() => {
+    INPUT_MIN_PAYOUTS.classList.remove('success');
+    ERROR_MESSAGE_ELEMENT.textContent = ``;
+    MODAL.close();
+  }, 1000);
+}
+
+function handleError(error) {
+  const errorMessage = error.message || error;
+  const sanitizedMessage = errorMessage.replace(
+    /because Kind is "min_payout".*/,
```

Comment:
Это для чего?

Reply:
Не могу понять, тут предусмотрено то что с бека может прийти несколько ошибок?

Reply:
c сервера приходит массив ошибок, я взял первую, все ошибки приходят в таком формате, ошибка + because Kind is "min_payout я  эту часть убрал и показываю только текст ошибки 
![Снимок экрана 2024-06-19 в 17 31 09](https://github.com/nnslvp/alephium-pool/assets/74107101/c94063b9-5ce7-4b11-8600-e2ff6d86109e)
<img width="574" alt="Снимок экрана 2024-06-19 в 17 32 13" src="https://github.com/nnslvp/alephium-pool/assets/74107101/f98d1230-d62a-4841-8814-27d34d0d2c02">